src/modules/tradeFormValidation/pure/TradeFormButtons/tradeButtonsMap.tsx

```diff
@@ -0,0 +1,146 @@
+import React from 'react'
+
+import { Trans } from '@lingui/macro'
+import styled from 'styled-components/macro'
+
+import { CompatibilityIssuesWarning } from 'modules/trade/pure/CompatibilityIssuesWarning'
+import { TradeLoadingButton } from 'modules/trade/pure/TradeLoadingButton'
+
+import { GpQuoteErrorCodes } from 'api/gnosisProtocol/errors/QuoteError'
+import { TradeApproveButton } from 'common/containers/TradeApprove'
+import { TokenSymbol } from 'common/pure/TokenSymbol'
+
+import { TradeFormButtonContext, TradeFormValidation } from '../../types'
+import { TradeFormBlankButton } from '../TradeFormBlankButton'
+
+interface ButtonErrorConfig {
+  text: JSX.Element | string
+  id?: string
+}
+
+interface ButtonCallback {
+  (context: TradeFormButtonContext): JSX.Element | null
+}
+
+const CompatibilityIssuesWarningWrapper = styled.div`
+  margin-top: -10px;
+`
+
+const quoteErrorTexts: Record<GpQuoteErrorCodes, string> = {
+  [GpQuoteErrorCodes.UNHANDLED_ERROR]: 'Error loading price. Try again later.',
+  [GpQuoteErrorCodes.TransferEthToContract]:
+    'Buying native currency with smart contract wallets is not currently supported',
```

Comment:
I think this is no longer a problem.
Will check with the backend.

Reply:
No longer a problem \o/

https://cowservices.slack.com/archives/C0375NV72SC/p1685712392155839

I would do it in a different PR though just to isolate and make the testing scope smaller